browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -844,13 +843,41 @@ IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringSimple) {
   content::WebContents* contents =
       browser()->tab_strip_model()->GetActiveWebContents();
 
-  ASSERT_EQ(true,
-            EvalJs(contents, "checkSelector('#ad-banner', 'display', 'none')"));
-
-  ASSERT_EQ(true, EvalJs(contents,
-                         "checkSelector('.ad-banner', 'display', 'block')"));
-
-  ASSERT_EQ(true, EvalJs(contents, "checkSelector('.ad', 'display', 'none')"));
+  auto result_first = EvalJsWithManualReply(contents,
+                                            R"(function waitCSSSelector() {
+          if (checkSelector('#ad-banner', 'display', 'none')) {
```

Comment:
In most of these tests, `checkSelector` returns a `Promise` (which evaluates to `true` in an `if` statement)... so these tests are not actually doing anything :frowning: 

It should be like this:

```js
(async function waitCSSSelector() {
    if (await checkSelector('#ad-banner', 'display', 'none')) {
    // ...
```